File: Python/Searching Algorithms/BinarySearch.py
def binarySearch():
    List = [23, 48, 54, 67, 71, 89]
    target = 23
    start = 0
    end = len(List) - 1
    while(start <= end):
        mid = start + (end - start) // 2
        if(List[mid] < target):
            start = mid + 1
        elif(List[mid] > target):
            end =  mid - 1
        else:
            return mid
    return -1

def main():
    result = binarySearch()
    if result != -1:
        print(f"Element found at index {result}")
    else:
        print("Element not found")
    
if __name__ == "__main__":
    main()

File: Java/Recursion/BinarySearch.java
package Recursion;

public class BinarySearch {
        int binarySearch(int arr[], int left, int right, int x){
           if (right >= left){
              int mid = left + (right - left) / 2;
              if (arr[mid] == x)
              return mid;
              if (arr[mid] > x)
              return binarySearch(arr, left, mid - 1, x);
              return binarySearch(arr, mid + 1, right, x);
           }
           return -1;
        }
        public static void main(String args[]){
           BinarySearch bs = new BinarySearch();
           int arr[] = { 56, 78, 90, 32, 45, 99, 104};
           int len = arr.length;
           int x = 104;
           int result = bs.binarySearch(arr, 0, len - 1, x);
           if (result == -1)
              System.out.println("The element is not present in the array");
           else
              System.out.println("The element has been found at index " + result);
        }
}
